### ISSUE TITLE
Add custom alias for app names

### ISSUE THREAD

**Issue by miguxbe** [flatpak/flatpak#103]
I've been playing with flatpak for a couple of weeks and one thing that I find annoying is constantly type long app names like org.gnome.Calendar. 

My question, is possbile to include some alias method to fix this? i.e org.gnome.Calendar -> alias: Calendar.  I mean, not automagical alieses but yes custom aliases.

**Comment by matthiasclasen** [comment]
The primary way we expect you to interact with desktop apps packaged as flatpaks is via your desktop shell. That is why we install desktop files that lauch them. Does that not work for you ? It is possible that you are not using gdm, and thus don't get the necessary XDG_DATA_DIRS setup for your desktop shell to show the flatpak apps. What is your display manager ?

As for aliases, I think the commandline completion should help you. Typing

flatpak run o<tab>g<tab>c<tab>

should give you a reasonably quick way to launch e.g. org.gnome.Calendar

**Comment by matthiasclasen** [comment]
err, what I meant to write was

flatpak run oTABgTABcTAB

**Comment by miguxbe** [comment]
OK, thanks for the clue. The problem is Flatpak completion works only under bash, I guess. I use ZSH and my completion suggest dirs instead. So I don't know how hard could be support ZSH or would be better create a ZSH plugin.  

BTW, I use GDM ;-)

**Comment by matthiasclasen** [comment]
I don't know anything about zsh completion, unfortunately. The bash completion support is now in the binary itself, so maybe a person with knowledge of zsh could come up with the required glue.

**Comment by xasx** [comment]
We could also have the issue handled in the zsh-users/zsh-completions repo, where the experts are.

Any thoughts?

**Comment by mvaello** [comment]
I agree. Is there any way to have a "cross" open issue? In order to give a context for zsh devs.

**Comment by mrflow** [comment]
Here's a quick and dirty solution, suitable for copy&pasting into .zshrc. It does not handle a few cases from the bash version that I didn't really try to understand:

```
_flatpak() {
    read -l
    local cl="$REPLY"
    read -ln
    local cp="$REPLY"
    local cur=""

    for ((i=${cp}-1; i>=0; --i)); do
        [[ x${cl[$i,$i]}x == "x x" ]] && cur=${cl[$i+1,$cp-1]} && break;
    done

    reply=($(flatpak complete "${cl}" "${cp}" "${cur}"))
}

compctl -K _flatpak flatpak
```

Works good enough for me.

**Comment by bil-elmoussaoui** [comment]
I've created a simple plugin for that https://github.com/bil-elmoussaoui/flatpak-zsh-completion ;) not that complete, need some more work ! but it's a start :+1:

**Comment by alex285** [comment]
i will put this issue here, so maybe someone could help whats should be the default behavior?
https://github.com/bil-elmoussaoui/flatpak-zsh-completion/issues/1

**Comment by tmccombs** [comment]
Here is an implementation of zsh completion that behaves very similarly to the bash_completion (and uses the new style of zsh completion):

```zsh
#compdef flatpak

setopt noshglob

local index
(( index = ${(c)#words[0,CURRENT]} + $#PREFIX + 1 ))


local resp=($(flatpak complete "$words" $index "$words[CURRENT]"))

_description options opt_expl option
_description arguments arg_expl argument

local match
for match in $resp; do
  case $match in
    __FLATPAK_FILE) _files;;
    __FLATPAK_BUNDLE_FILE) _path_files -g '*.flatpak';;
    __FLATPAK_BUNDLE_OR_REF_FILE) _path_files -g '*.flatpak(|ref)';;
    __FLATPAK_DIR) _path_files -/;;
    -*=) compadd $opt_expl[@] -S "" -- $match;;
    -*) compadd $opt_expl[@] -- $match;;
    *) compadd $arg_expl[@] $match;;
  esac
done
```

**Comment by mati865** [comment]
@tmccombs maybe you could open PR to this repository or [`zsh-completions`](https://github.com/zsh-users/zsh-completions/)?
